detox/src/devices/drivers/InstrumentationLogsParser.js

```diff
@@ -0,0 +1,42 @@
+const INSTRUMENTATION_LOGS_PREFIX = 'INSTRUMENTATION_STATUS';
+const STACKTRACE_PREFIX_TEXT = INSTRUMENTATION_LOGS_PREFIX + ': stack=';
+
+class InstrumentationLogsParser {
+  hasStackTraceLog(logsDump) {
+    return logsDump.includes(STACKTRACE_PREFIX_TEXT);
+  }
+
+  getStackTrace(logsDump) {
+    const logLines = logsDump.split('\n');
+
+    const index = this._findStackTraceLog(logLines);
+    const stackTrace = this._extractStackTrace(logLines, index);
+    return stackTrace;
+  }
+
+  _findStackTraceLog(logLines) {
```

Comment:
Can be shortened to:

```js
const PREFIX = 'INSTRUMENTATION_STATUS';
const STARTING_WITH_PREFIX = new RegExp(`^${PREFIX}`, 'm');
const STACKTRACE_SUFFIX = ': stack=';
const STACKTRACE_MARKER = PREFIX + STACKTRACE_SUFFIX;

class InstrumentationLogsParser {
  hasStackTraceLog(logsDump) {
    return logsDump.includes(STACKTRACE_MARKER);
  }

  getStackTrace(logsDump) {
    const stackStatus = logsDump
      .split(STARTING_WITH_PREFIX)
      .find(status => status.startsWith(STACKTRACE_SUFFIX)) || '';

    return stackStatus.slice(STACKTRACE_SUFFIX.length);
  }
}
```

This code also passes your unit tests.

Reply:
amazing